### PR TITLE
Add a new method to call the contract

### PR DESCRIPTION
**Pull request by Shareong**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -66,19 +66,17 @@ List alibaba = [
...
-	compile 'org.apache.commons:commons-lang3:3.1'
-	compile 'com.fasterxml.jackson.core:jackson-databind:2.9.9.2'
-	compile 'io.netty:netty-all:4.1.32.Final'
-	compile 'io.netty:netty-tcnative:2.0.25.Final'
-	compile 'io.netty:netty-tcnative-boringssl-static:2.0.20.Final'
-	compile 'com.google.guava:guava:28.1-jre'
-	compile 'commons-configuration:commons-configuration:1.10'
-	 // web3j
-	compile 'org.apache.httpcomponents:httpclient:4.5.5',
+    compile 'org.ethereum:solcJ-all:0.4.25'
```
Comment (ywy2090) [inline build.gradle]: Should add solcJ-all?

File: src/main/java/org/fisco/bcos/channel/client/CallContract.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        return receipt.getOutput();
+    }
+
+    public List<Type> decode(String data, TypeReference<?>... typeReferences) {
+        List<TypeReference<?>> typeReferencesList = Arrays.<TypeReference<?>>asList(typeReferences);
+        return FunctionReturnDecoder.decode(data, Utils.convert(typeReferencesList));
+    }
+
+    @SuppressWarnings("unchecked")
+    public <S extends Type, T> List<T> convertList(List<S> arr) {
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/channel/client/CallContract.java]: convertList already defined in Contract.java, move it to Utils.java as common tools will be better

File: src/main/java/org/fisco/bcos/channel/client/CallContract.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        List<T> out = new ArrayList<T>();
+        for (Iterator<S> it = arr.iterator(); it.hasNext(); ) {
+            out.add((T) it.next().getValue());
+        }
+        return out;
+    }
+
+    public <S extends Type, T> List<List<T>> convertListList(List<StaticArray<S>> arrs) {
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/channel/client/CallContract.java]: The same as convertList. Define it in Utils.java

File: src/main/java/org/fisco/bcos/channel/client/CallContract.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        Call ethCall =
+                web3j.call(
+                                Transaction.createEthCallTransaction(
+                                        credentials.getAddress(), contractAddress, data),
+                                DefaultBlockParameterName.LATEST)
+                        .send();
+        return ethCall.getValue().getOutput();
+    }
+
+    public String sendTransaction(
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/channel/client/CallContract.java]: Add another sendTransaction interface without gas, gasPrice using default value will be better .

File: src/main/java/org/fisco/bcos/channel/client/CallContract.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                        funcName,
+                        Arrays.<Type>asList(args),
+                        Collections.<TypeReference<?>>emptyList());
+        String data = FunctionEncoder.encode(function);
+        Call ethCall =
+                web3j.call(
+                                Transaction.createEthCallTransaction(
+                                        credentials.getAddress(), contractAddress, data),
+                                DefaultBlockParameterName.LATEST)
+                        .send();
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/channel/client/CallContract.java]: ethCall.getValue() maybe null ?

File: src/main/java/org/fisco/bcos/channel/client/CallContract.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                new Function(
+                        funcName,
+                        Arrays.<Type>asList(args),
+                        Collections.<TypeReference<?>>emptyList());
+        SendTransaction sendTransaction =
+                new SendTransaction(contractAddress, web3j, credentials, gasPrice, gasLimit);
+        TransactionCallback callback = new TransactionCallback();
+        try {
+            sendTransaction.send(function, callback);
+        } catch (Exception e) {
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/channel/client/CallContract.java]: Contract.java' interface executeTransaction ?

File: src/main/java/org/fisco/bcos/channel/client/CallContract.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+            Thread.currentThread().interrupt();
+            logger.error("error:", e);
+        }
+
+        TransactionReceipt receipt = callback.receipt;
+        return receipt.getOutput();
+    }
+
+    public List<Type> decode(String data, TypeReference<?>... typeReferences) throws Exception {
+        if (data.isEmpty()) throw new Exception("input is null");
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/channel/client/CallContract.java]: data.isEmpty() or data.equals("0x")

File: src/main/java/org/fisco/bcos/channel/client/CallContract.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+                        .send();
+        Call.CallOutput callOutput = ethCall.getValue();
+        if (callOutput != null) {
+            return callOutput.getOutput();
+        } else {
+            return null;
+        }
+    }
+
+    public String sendTransaction(String contractAddress, String funcName, Type... args) {
```
Comment (JimmyShi22) [inline src/main/java/org/fisco/bcos/channel/client/CallContract.java]:
```suggestion
    public String sendRawTransaction(String contractAddress, String funcName, Type... args) {
```
It's better to use the same function as we known

**Pull request merged**